### PR TITLE
Fixes #1 Uses click to accept path for proposed endpoints doc

### LINKED ISSUE #1: Allow `serialize_data.py` accept path for proposed endpoints docs as command line argument
The `serialize_data.py` file currently has `proposed_endpoints.md` file hard-coded.

Modify this file to accept a path for proposed endpoints docs as command line argument.

This is a beginner-friendly issue.

### PR DESCRIPTION
**Pull request by mansiag**
Uses click to Allow `serialize_data.py` to accept path for proposed endpoints docs as command line argument.

### CODE REVIEW

File: src/serialize_data.py [hunk 1]
```diff
@@ -93,9 +94,11 @@ def get_json_from_endpoints(lines):
     return endpoint_json_list
 
 
-def generate_json_from_docs_file(filename='proposed_endpoints.md'):
+@click.command()
+@click.argument('file_path', type=click.Path(exists=True))
+def generate_json_from_docs_file(file_path):
```
Comment (CuriousLearner) [inline src/serialize_data.py:99]: I believe that this path shouldn't be mandatory to provide. What do you think?

Comment (mansiag) [reply src/serialize_data.py:99]: Yes, its better to have some default value of the path. I would make the necessary changes.

**Comment by CuriousLearner** [comment]
Last few things:

 - Document the feature
 - Add changelog
 - Fix the CI.

**Comment by mansiag** [comment]
Can you suggest how should I document the feature in Readme

**Comment by CuriousLearner** [comment]
State how do we execute the command with `--file_path` option.

**Comment by mansiag** [comment]
@CuriousLearner please review.

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ pipenv install
 
 ## Usage
 
-Run `pipenv run python serialize_data.py` to generate JSON file named `endpoints_data.json`
+Run `pipenv run python serialize_data.py` to generate JSON file named `endpoints_data.json`. One can specify the path of the proposed endpoints docs  using `--file_path` option in the above command.
```
Comment (CuriousLearner) [inline README.md]: Ideally command line option should use `-` hyphens instead of underscore `_`

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+
+2019-1-29
+=======================
+
+Features & Improvements
+-----------------------
+
+- Allow `serialize_data.py` accept path for proposed endpoints docs as command line argument. [#1](https://github.com/CuriousLearner/API-stub-generator/pull/4)
```
Comment (CuriousLearner) [inline CHANGELOG.md]: Instead of URL to the issue, you can add your name and link to your GitHub profile.

**Comment by CuriousLearner** [comment]
@mansiag 
Just minor nit-picks, rest looks good to me :)

File: README.md (hunk 2 above)
Comment (mansiag) [reply README.md]: But it will give syntax error on using file-path as an argument to the generate_json_from_docs_file function.

Comment (CuriousLearner) [reply README.md]: @mansiag I think there should be a way, to use `-` as part of the actual argument that receives the callback but also have the actual value come in as a valid identifier (that uses `_` to be consistent with the Python naming conventions).

Can you try finding a way?

Comment (mansiag) [reply README.md]: Yeah Done.

**Review by CuriousLearner — APPROVED** [review]
Hey, @mansiag Thanks for your work. I've handled minor documentation thing and it seems good to go!

Thanks for your patience and all your work :)

**Pull request merged**